client/src/components/categorical/occupancy.js

```diff
@@ -0,0 +1,85 @@
+// jshint esversion: 6
+import React from "react";
+import _ from "lodash";
+import { connect } from "react-redux";
+import * as d3 from "d3";
+
+@connect()
+class Occupancy extends React.Component {
+  render() {
+    const {
+      occupancy,
+      colorScale,
+      categoricalSelectionState,
+      colorAccessor,
+      schema
+    } = this.props;
+    const width = 100;
+    const height = 11;
+
+    const categories = _.filter(schema.annotations.obs, {
+      name: colorAccessor
+    })[0].categories;
+
+    const x = d3
+      .scaleLinear()
+      /* get all the keys d[1] as an array, then find the sum */
+      .domain([0, d3.sum(Array.from(occupancy, d => d[1]))])
+      .range([0, width]);
+
+    const stack = [];
+    let currentOffset = 0;
+
+    // occupancy.forEach((value, key) => {
+    //   const scaledValue = x(value);
+    //   stack.push({
+    //     key,
+    //     value,
+    //     rectWidth: scaledValue,
+    //     offset: currentOffset,
+    //     fill: colorScale(categories.indexOf(key))
+    //   });
+    //   currentOffset += scaledValue;
+    // });
+
+    const cat = categoricalSelectionState[colorAccessor];
+
+    cat.categoryValues.forEach(d => {
```

Comment:
this could be rewritten as `cat.categoryValues.map(...)`, and get rid of the manual stack.push().  Style comment only, so at your discretion.  I tend to prefer map() when that is the only side effect.